src/uv_snapshot.c

```diff
@@ -578,6 +578,11 @@ static void uvSnapshotPutAfterWorkCb(uv_work_t *work, int status)
     struct uv *uv = put->uv;
     assert(status == 0);
     uv->snapshot_put_work.data = NULL;
+    /* The next entry in the first open segment will come after the snapshot
+     * entries. */
+    if (put->trailing == 0 && uv->prepare_next_counter == 0) {
+        uv->prepare_next_counter = 1;
```

Comment:
It might be unrelated to the Jenkins failures, but after a very quick look at the PR's diff I noticed this change, and I'm not entirely sure it's enough. Could it be that there is a ready but yet unused `open-0` open segment that will be the one used after the snapshot is installed?

Just throwing it there in case it help, didn't really look much.